client/src/js/directives/findDebtorGroup.js

```diff
@@ -0,0 +1,117 @@
+/**
+* <bh-find-debtorgroup></bh-find-debtorgroup>
+*
+* The find debtor group component is responsible for displaying a component
+* for searching a debtor group or a convention
+*
+* This component takes these attributes :
+*   - is-onvention : if we want only debtor groups which are conventions
+*   - on-search-complete : the callback function which get the returned debtror group
+*
+* @example
+* // component find debtor group
+* <bh-find-debtorgroup>
+*   is-convention="true"
+*   on-search-complete="MyCtrl.handleResult"
+* </bh-find-debtorgroup>
+*/
+
+/** Find Debtor Group Controller */
+function findDebtorGroupController($http) {
+  var ctrl         = this,
+      callback     = this.callback,
+      locked       = this.locked,
+      isConvention = this.isConvention;
+
+  /** Load debtor group */
+  loadgDebtorGroup();
+
+  /**
+  * loading state :
+  * This variable takes these values (0|1|2) :
+  *   - 0 = loading
+  *   - 1 = loaded
+  *   - 2 = error
+  *   - null = either value
+  */
+  ctrl.state = null;
+
+  /**
+  * Search view
+  */
+  ctrl.showSearchView = 'default';
+
+  ctrl.selectDebtorGroup = function(debtorGroup) {
+    if (!debtorGroup) { ctrl.state = 2; }
+    ctrl.selected = debtorGroup;
+    ctrl.state = 1;
+    ctrl.showSearchView = 'result';
+    callback({ debtorGroup: debtorGroup });
+  };
+
+  ctrl.reload = function() {
+    ctrl.state = null;
+    ctrl.showSearchView = 'default';
+  };
+
+  /**
+  * @function loadgDebtorGroup
+  *
+  * @desc This function is responsible to loading all debtor groups
+  *
+  * @param {string} name The name of the debtor group
+  */
+  function loadgDebtorGroup() {
+    ctrl.state = 0;
+    var criterion = {};
+
+    /** Is convention debtor group or not */
+    if (angular.isDefined(isConvention) && Boolean(isConvention) === true) {
+      criterion.is_convention = 1;
+    }
+    else if (angular.isDefined(isConvention) && Boolean(isConvention) === false) {
+      criterion.is_convention = 0;
+    }
+
+    /** Locked debtor group or not */
+    if (angular.isDefined(locked) && Boolean(locked) === true) {
```

Comment:
Instead of checking if these values are defined each time, why not default them to a value?  For example
```js
function findDebtorGroupController($http) {
  var ctrl = this;

  //  ... code ...
 ctrl.locked = ctrl.locked || false;
 ctrl.isConvention = ctrl.isConvention || false;

  // no need to check angular.isDefined() anymore!
});
```